Helper/PaymentMethods.php

```diff
@@ -400,38 +309,111 @@ protected function getCurrentShopperReference()
     }
 
     /**
-     * @return array|mixed
-     * @throws \Adyen\AdyenException
+     * @param $merchantAccount
+     * @param \Magento\Store\Model\Store $store
+     * @param $country
+     * @param \Magento\Quote\Model\Quote $quote
+     * @return array
+     * @throws \Exception
      */
-    public function getConnectedTerminals()
-    {
-        $storeId = $this->session->getQuote()->getStoreId();
+    protected function getPaymentMethodsRequest(
+        $merchantAccount,
+        \Magento\Store\Model\Store $store,
+        $country,
+        \Magento\Quote\Model\Quote $quote
+    ) {
+        $paymentMethodRequest = [
+            "channel" => "Web",
+            "merchantAccount" => $merchantAccount,
+            "countryCode" => $this->getCurrentCountryCode($store, $country),
+            "shopperLocale" => $this->adyenHelper->getCurrentLocaleCode($store->getId())
+        ];
 
-        // initialize the adyen client
-        $client = $this->adyenHelper->initializeAdyenClient($storeId, $this->adyenHelper->getPosApiKey($storeId));
+        if (!empty($this->getCurrentShopperReference())) {
+            $paymentMethodRequest["shopperReference"] = $this->getCurrentShopperReference();
+        }
+        $currencyCode = $this->getCurrentCurrencyCode($store);
+        $paymentMethodRequest["amount"]["currency"] = $currencyCode;
```

Comment:
```suggestion
```

Reply:
I have to use the  $currencyCode for the format amount:  
```$amountValue = $this->adyenHelper->formatAmount($this->getCurrentPaymentAmount(), $currencyCode);```
That is why I set it later